src/sql/src/parsers/tql_parser.rs

```diff
@@ -0,0 +1,205 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use snafu::ResultExt;
+use sqlparser::keywords::Keyword;
+use sqlparser::parser::ParserError;
+use sqlparser::tokenizer::Token;
+
+use crate::error::{self, Result};
+use crate::parser::ParserContext;
+use crate::statements::statement::Statement;
+use crate::statements::tql::{Tql, TqlEval, TqlExplain};
+
+pub const TQL: &str = "TQL";
+const EVAL: &str = "EVAL";
+const EXPLAIN: &str = "EXPLAIN";
+use sqlparser::parser::Parser;
+
+/// TQL extension parser, including:
+/// - TQL EVAL <query>
+/// - TQL EXPLAIN <query>
+impl<'a> ParserContext<'a> {
+    pub(crate) fn parse_tql(&mut self) -> Result<Statement> {
+        self.parser.next_token();
+
+        match self.parser.peek_token() {
+            Token::Word(w) => match w.keyword {
+                Keyword::NoKeyword if w.value.to_uppercase() == EVAL && w.quote_style.is_none() => {
+                    self.parser.next_token();
+                    self.parse_tql_eval()
+                        .context(error::SyntaxSnafu { sql: self.sql })
+                }
+
+                Keyword::EXPLAIN => {
+                    self.parser.next_token();
+                    self.parse_tql_explain()
+                }
+
+                _ => self.unsupported(self.peek_token_as_string()),
+            },
+            unexpected => self.unsupported(unexpected.to_string()),
+        }
+    }
+
+    fn parse_tql_eval(&mut self) -> std::result::Result<Statement, ParserError> {
+        let parser = &mut self.parser;
+        parser.expect_token(&Token::LParen)?;
+        let start = Self::parse_string_or_number(parser, Token::Comma)?;
+        let end = Self::parse_string_or_number(parser, Token::Comma)?;
+        let step = Self::parse_string_or_number(parser, Token::RParen)?;
+        let query = Self::parse_tql_query(parser, self.sql, ")")?;
+
+        Ok(Statement::Tql(Tql::Eval(TqlEval {
+            start,
+            end,
+            step,
+            query,
+        })))
+    }
+
+    fn parse_string_or_number(
+        parser: &mut Parser,
+        token: Token,
+    ) -> std::result::Result<String, ParserError> {
+        let value = match parser.next_token() {
+            Token::Number(n, _) => n,
+            Token::DoubleQuotedString(s) | Token::SingleQuotedString(s) => s,
+            unexpected => {
+                return Err(ParserError::ParserError(format!(
+                    "Expect number or string, but is {unexpected}"
```

Comment:
debug output to carry type:

```suggestion
                    "Expect number or string, but is {unexpected:?}"
```